predicthq/endpoints/decorators.py

```diff
@@ -16,16 +16,25 @@ def _to_url_params(data, glue=".", separator=","):
         elif isinstance(value, list):
             params[key] = separator.join(map(str, value))
         elif isinstance(value, dict):
-            for subkey, subvalue in value.items():
-                if isinstance(subvalue, list):
-                    params[glue.join((key, subkey))] = separator.join(map(str, subvalue))
-                else:
-                    params[glue.join((key, subkey))] = subvalue
+            params.update(**_flatten_dict(value, glue, separator, pk=key))
         else:
             params[key] = value
     return params
 
 
+def _flatten_dict(d, glue, separator, pk=""):
```

Comment:
let's use more explicit var names for easier readability, e.g. does pk mean primary key? or parent key?

Reply:
Done